realm/realm-library/src/androidTest/java/io/realm/LinkingObjectsManagedTests.java

```diff
@@ -0,0 +1,741 @@
+        parent.getFieldList().add(child);
+        realm.commitTransaction();
+
+        assertEquals(1, child.getListParents().size());
+        assertTrue(child.getListParents().contains(parent));
+    }
+
+    // Setting multiple object links creates multiple backlinks
+    @Test
+    public void basic_multipleBacklinksObject() {
+        realm.beginTransaction();
+        AllJavaTypes child = realm.createObject(AllJavaTypes.class, 1);
+        AllJavaTypes parent1 = realm.createObject(AllJavaTypes.class, 2);
+        AllJavaTypes parent2 = realm.createObject(AllJavaTypes.class, 3);
+        parent1.setFieldObject(child);
+        parent2.setFieldObject(child);
+        realm.commitTransaction();
+        assertEquals(2, child.getObjectParents().size());
+    }
+
+    // Setting multiple list links creates multiple backlinks
+    @Test
+    public void basic_multipleBacklinksList() {
+        realm.beginTransaction();
+        AllJavaTypes child = realm.createObject(AllJavaTypes.class, 1);
+        AllJavaTypes parent1 = realm.createObject(AllJavaTypes.class, 2);
+        AllJavaTypes parent2 = realm.createObject(AllJavaTypes.class, 3);
+        parent1.getFieldList().add(child);
+        parent2.getFieldList().add(child);
+        realm.commitTransaction();
+        assertEquals(2, child.getListParents().size());
+    }
+
+    // Adding multiple list links creates multiple backlinks,
+    // even if the links are to a single object
+    // !!!FIXME: DOCUMENT THIS
+    @Test
+    public void basic_multipleReferencesFromParentList() {
+        realm.beginTransaction();
+        AllJavaTypes child = realm.createObject(AllJavaTypes.class, 1);
+        AllJavaTypes parent = realm.createObject(AllJavaTypes.class, 2);
+        parent.getFieldList().add(child);
+        parent.getFieldList().add(child);
+        realm.commitTransaction();
+
+        // One entry for each reference, so two references from a LinkList will
+        // result in two backlinks.
+        assertEquals(2, child.getListParents().size());
+        assertEquals(parent, child.getListParents().first());
+        assertEquals(parent, child.getListParents().last());
+    }
+
+    // A listener registered on the backlinked object should be called when a commit adds a backlink
+    @Test
+    @RunTestInLooperThread
+    public void notification_onCommitModelObject() {
+        final Realm looperThreadRealm = looperThread.realm;
+
+        looperThreadRealm.beginTransaction();
+        AllJavaTypes child = looperThreadRealm.createObject(AllJavaTypes.class, 10);
+        looperThreadRealm.commitTransaction();
+
+        final AtomicInteger counter = new AtomicInteger(0);
+        RealmChangeListener<AllJavaTypes> listener = new RealmChangeListener<AllJavaTypes>() {
+            @Override
+            public void onChange(AllJavaTypes object) {
+                counter.incrementAndGet();
+            }
+        };
+        child.addChangeListener(listener);
```

Comment:
I don't think that is right.  `PostConditions.run` is called after the test method is gone.  Although it is unlikely that the GC will happen right then, it could.  If it does, there are no references to those objects and they will be GCed.